tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4207,6 +4210,57 @@ TEST_F(ValidationTest, ValidateWithHash) {
   VERIFY_ARE_EQUAL(memcmp(Result, pHeader->Hash.Digest, sizeof(Result)), 0);
 }
 
+TEST_F(ValidationTest, ValidateWithExternalValidator) {
+  /*
+  // The below part of the test was run on a local machine,
+  // but shouldn't be run yet in automated scenarios, since
+  // the DXC repo does not yet have a checked-in dxil.dll.
+
+  // set the environment variable that stores the path to the extenral
+  // dxil.dll
+  SetEnvironmentVariableW(L"DXC_DXIL_DLL_PATH",
+                          L"D:\\hlsl.bin\\Debug\\bin\\dxil.dll");
+  // also update the CRT environment
+  _putenv_s("DXC_DXIL_DLL_PATH", "D:\\hlsl.bin\\Debug\\bin\\dxil.dll");
+  */
+
+  if (!m_dllExtSupport.IsEnabled()) {
+    VERIFY_SUCCEEDED(m_dllExtSupport.Initialize());
+  }
+
+  if (m_ver.SkipDxilVersion(1, ShaderModel::kHighestReleasedMinor))
+    return;
+  CComPtr<IDxcBlob> pProgram;
+  CompileSource("float4 main(float a:A, float b:B) : SV_Target { return 1; }",
+                "ps_6_0", &pProgram);
+
+  CComPtr<IDxcValidator> pValidator;
+  CComPtr<IDxcOperationResult> pResult;
+  unsigned Flags = 0;
+  VERIFY_SUCCEEDED(
+      m_dllExtSupport.CreateInstance(CLSID_DxcValidator, &pValidator));
+  // With hash.
+  VERIFY_SUCCEEDED(pValidator->Validate(pProgram, Flags, &pResult));
+  // Make sure the validation was successful.
+  HRESULT status;
+  VERIFY_IS_NOT_NULL(pResult);
+  CComPtr<IDxcBlob> pValidationOutput;
+  pResult->GetStatus(&status);
+  VERIFY_SUCCEEDED(status);
+  pResult->GetResult(&pValidationOutput);
+
+  /*
+  // as described above, this should only be tested in local scenarios,
+  // until dxil.dll is checked into the dxc repo
+
+  // validate that m_dllExtSupport was able to capture the environment
+  // variable's value, and that loading the external validator was successful
+  VERIFY_IS_TRUE(!m_dllExtSupport.DxilDllFailedToLoad());
```

Comment:
I think that the `DxcDllExtValidationSupport` being tested should be a local variable in the test method.

We don't need to compile and validate a shader to unit test certain aspects of the `DxcDllExtValidationSupport` class.  We aren't actually testing the compile behavior yet.

We should be able to unit test several things without compiling:
- bogus path set: is retrieved by `GetDxilDllPath()` and fails to load, returning true for `DxilDllFailedToLoad()`
- path is not set: `GetDxilDllPath()` is empty and `DxilDllFailedToLoad()` returns false
- you can successfully create instances of `CLSID_DxcCompiler`, `CLSID_DxcLinker`, `CLSID_DxcValidator` through `DxcDllExtValidationSupport`.